nxdrive/engine/dao/sqlite.py

```diff
@@ -2778,19 +2784,19 @@ def resume_session(self, uid: int) -> None:
         """Resume all transfers for given session."""
         with self.lock:
             c = self._get_write_connection().cursor()
+
+            # Resume uploads that were already started when the session was paused
             c.execute(
-                "UPDATE Uploads SET status = ? WHERE doc_pair IN (SELECT id FROM States WHERE Session = ?)",
+                "UPDATE Uploads SET status = ? WHERE doc_pair IN (SELECT id FROM States WHERE session = ?)",
                 (TransferStatus.ONGOING.value, uid),
             )
 
+            # Push not-yet-handled session transfers into the queue
             if self.queue_manager:
-                rows = c.execute(
-                    "SELECT * FROM States WHERE id IN"
-                    " (SELECT doc_pair FROM Uploads WHERE doc_pair IN (SELECT id FROM States WHERE Session = ?))",
-                    (uid,),
-                ).fetchall()
-                for doc_pair in rows:
+                rows = c.execute("SELECT * FROM States WHERE session = ?", (uid,))
```

Comment:
Hm I just found something here: we potentially repush the same `doc_pair` for all ongoing uploads. And duplicates may happen. Right?